src/oled_display_node.cpp

```diff
@@ -162,53 +165,48 @@ extern  int dispOled_writeText(dispCtx_t *dispCtx, uint8_t line, uint8_t segment
  * @note    This routine only uses the readable stream returned by `popen`
  */
 std::string getPopen(std::string input) {
-    std::string result("");;
-    char c;
-    FILE* fp;
-
-    try {   // Catch any bad operations with this trap
-
-    fp = popen(input.c_str(), "r");
-    if (fp != NULL) {
-        do {
-            c = fgetc(fp);
-            if (isascii(c)) {
-                result += c;
-            }
-        } while (!feof(fp));
-        pclose(fp);
-    }
+        std::string result("");;
+        char c;
+        FILE* fp;
+
+        try {   // Catch any bad operations with this trap
+
+        fp = popen(input.c_str(), "r");
+        if (fp != NULL) {
+                do {
+                        c = fgetc(fp);
+                        if (isascii(c)) {
+                                result += c;
+                        }
+                } while (!feof(fp));
+                pclose(fp);
+        }
 
-    } catch (...) {  }
+        } catch (...) {  }
 
-    return result;
+        return result;
 }
 
-// getMainIpAddress()
-// We will use any eth0 IP address that has 192.168.1 in it unless that is not found in which case use wlan0 IP
-// For some systems with Lidar the ethernet IP is 192.168.42.x  so in that case wlan0 will be used rather than hard lan IP.
-// One disadvantage is on older images we did not rename the ethernet to enet0 so we will not get those plugged in
-// network cables but this is a robot and plugged in lan is not the typical usage, wlan0 is 'normal'
-//
-void  getMainIpAddress(std::string &ipAddress, int logFindings) {
-  // ip -o -4 addr returns:  2: eth0    inet 192.168.1.164/24 brd 192.168.1.255 scope global eth0\       valid_lft forever 
-  std::string  enetIpAddress = getPopen("ip -o -4 addr | grep eth0  | awk '{print $4}'");  // IP with /24 mask bits at end
-  std::string  wlanIpAddress = getPopen("ip -o -4 addr | grep wlan0 | awk '{print $4}'");  // IP with /24 mask bits at end
-  std::string  mainIpAddress("notfound");
-
-  if (enetIpAddress.length() > 8) {
-      mainIpAddress.assign(enetIpAddress.substr(0,enetIpAddress.find('/')));
-      if (logFindings != 0) {
-          ROS_INFO("%s The eth0 IP will be the displayed IP of %s", THIS_NODE_NAME, mainIpAddress.c_str());
-      }
-  } else {
-      mainIpAddress.assign(wlanIpAddress.substr(0,wlanIpAddress.find('/')));
-      if (logFindings != 0) {
-          ROS_INFO("%s The wlan0 IP will be the displayed IP of %s", THIS_NODE_NAME, mainIpAddress.c_str());
-      }
-  }
-  ipAddress.assign(mainIpAddress);
-  return;
+// getIpAddressses()
+// We show both the WLAN0 and ETH0 adapter IPs since both can be required by the user
```

Comment:
I will prey to the gods for this iteration.   My own tests look rather good so far but I bet there is SOME combination that still is not happy.  Still, this is a nice improvement